_posts/2018-05-02-load-balanced-smart-proxies-with-haproxy.md

```diff
@@ -0,0 +1,151 @@
+
+```yaml
+pulp::lazy_redirect_host: smartproxy.example.net
+```
+### Running the installer
+As only the first Smart Proxy will host a PuppetCA, we configure that explicitly during the installation and also configure the Puppet certificate to include the `smartproxy.example.net` name:
+```
+[root@smartproxy01] # foreman-installer \
+…
+--puppet-dns-alt-names smartproxy.example.net \
+--puppet-ca-server proxy01.example.net \
+--foreman-proxy-puppetca true \
+--puppet-server-ca true
+```
+
+## Installing the other Smart Proxies
+The installation of all other Smart Proxies is almost identical to the first one. We just have to prepare the Puppet certs on the first one (as it's the only one having a PuppetCA) and disable the PuppetCA in the installer.
+
+### Generating Katello certificates
+This step is identical to the steps used to generate the certificates for the first smart proxy. Simply use `foreman-proxy-certs-generate` as described above.
+
+### Generating Puppet certificates
+On the first smart proxy, generate the Puppet certificates by running
+
+```shell
+puppet cert generate proxy02.example.net --dns_alt_names=smartproxy.example.net
+```
+
+Install the `puppetserver` package on the second smart proxy, to create the `puppet` user and the directory structure in `/etc/puppetlabs`.
+
+Now copy `/etc/puppetlabs/puppet/ssl/*/proxy02.example.net.pem` and `/etc/puppetlabs/puppet/ssl/certs/ca.pem` to the second smart proxy and ensure ownership and permissions are identical to those the first host. Also don't forget to check the SELinux contexts of the files you copied over (hint: `ls -lZ`) :)
+
+Repeat this step for all smart proxies that are going to sit behind the same load balancer.
+
+### Preparing `custom-hiera.yaml`
+These steps are identical to the steps used on the first Smart Proxy.
+
+### Running the installer
+As only the first Smart Proxy will host a PuppetCA, we point the Puppet agent to it explicitly during the installation and also configure the Puppet certificate to include the `smartproxy.example.net` name:
+```
+[root@smartproxy02] # foreman-installer \
+…
+--puppet-dns-alt-names smartproxy.example.net \
+--puppet-ca-server proxy01.example.net \
+--foreman-proxy-puppetca false \
+--puppet-server-ca false
+```
+## Installing HAProxy
+
+Please note that if you are using a different load balancer technology, you can use this configuration as reference.
+
+Configure SELinux to run in `permissive` mode.
+
+The HAProxy setup itself is super simple, just load-balance a bunch of ports in TCP mode round-robin across all the Smart Proxies. TCP mode is required because the load balancer will not be doing TLS termination. There are just two exceptions to that: port 443 should use `source` not `roundrobin` to not confuse Yum, and the additional port for the PuppetCA (8141) is forwarded only to the first smart proxy.
```

Comment:
I read somewhere that you should avoid describing things as simple. It can make beginners feel stupid when they think it's not simple. Maybe it's better to use `straight forward`?